### PR TITLE
chore: Add devcontainer config for Kvrocks

### PR DESCRIPTION
**Pull request by what-is-me**
Reference: https://github.com/apache/kvrocks/issues/2613

### CODE REVIEW

**Comment by git-hulk** [comment]
No worry about the CI failure, it's not related to this change.

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+FROM ubuntu:latest
+RUN apt update \
+    && apt install -y \
+        git build-essential cmake libtool python3 libssl-dev python3-pip \
+        wget curl clang-format-14 clang-tidy-14 golang-go ninja-build redis-tools vim
+RUN echo "export NPROC=$(nproc)" >> $HOME/.bashrc
+RUN apt install -y python3-redis redis-server
```
Comment (PragmaTwice) [inline .devcontainer/Dockerfile]:
```suggestion
RUN apt update \
    && apt install -y \
        git build-essential cmake libtool python3 libssl-dev python3-pip \
        wget curl clang-format-14 clang-tidy-14 golang-go ninja-build \
        redis-tools vim python3-redis redis-server clang lld mold gdb fish
```

File: .devcontainer/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,10 @@
+FROM ubuntu:latest
```
Comment (PragmaTwice) [inline .devcontainer/Dockerfile]:
```suggestion
FROM ubuntu:noble
```

Comment (PragmaTwice) [reply .devcontainer/Dockerfile]: I think we can pin to Ubuntu noble to prevent unexpected updates.

**Review by PragmaTwice — APPROVED** [review]

**Pull request merged**